### PR TITLE
KAFKA-15135: fix(storage): pass endpoint configurations as client commont to TBRLMM

### PR DESCRIPTION
**Pull request by jeqo**
Pass endpoint properties from RLM to TBRLMM and validate those are not ignored. 

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/test/java/kafka/log/remote/RemoteLogManagerTest.java [hunk 1]
```diff
@@ -220,8 +220,8 @@ void testRemoteLogMetadataManagerWithEndpointConfig() {
 
         ArgumentCaptor<Map<String, Object>> capture = ArgumentCaptor.forClass(Map.class);
         verify(remoteLogMetadataManager, times(1)).configure(capture.capture());
-        assertEquals(host + ":" + port, capture.getValue().get("bootstrap.servers"));
-        assertEquals(securityProtocol, capture.getValue().get("security.protocol"));
+        assertEquals(host + ":" + port, capture.getValue().get("remote.log.metadata.common.client.bootstrap.servers"));
+        assertEquals(securityProtocol, capture.getValue().get("remote.log.metadata.common.client.security.protocol"));
```
Comment (showuon) [inline core/src/test/java/kafka/log/remote/RemoteLogManagerTest.java]: Any reason why don't we use `REMOTE_LOG_METADATA_COMMON_CLIENT_PREFIX` like other places did?

**Review by showuon — COMMENTED** [review]
Thanks for the PR. Left a minor comment.

File: core/src/test/java/kafka/log/remote/RemoteLogManagerTest.java (hunk 1 above)
Comment (jeqo) [reply core/src/test/java/kafka/log/remote/RemoteLogManagerTest.java]: Sure, I was following the text content instead, but we can use the constant here as well. Applying fix

**Review by showuon — APPROVED** [review]
LGTM! Thanks for the fix! Let's wait for the CI build.

**Comment by showuon** [comment]
Failed tests are unrelated:
```
    Build / JDK 8 and Scala 2.12 / org.apache.kafka.connect.mirror.integration.MirrorConnectorsIntegrationTransactionsTest.testReplicationWithEmptyPartition()
    Build / JDK 8 and Scala 2.12 / org.apache.kafka.connect.integration.TransformationIntegrationTest.testFilterOnTopicNameWithSinkConnector
    Build / JDK 8 and Scala 2.12 / integration.kafka.server.FetchFromFollowerIntegrationTest.testRackAwareRangeAssignor()
    Build / JDK 8 and Scala 2.12 / kafka.admin.ResetConsumerGroupOffsetTest.testResetOffsetsShiftMinus()
    Build / JDK 8 and Scala 2.12 / kafka.admin.ResetConsumerGroupOffsetTest.testResetOffsetsByDurationFallbackToLatestWhenNoRecords()
    Build / JDK 8 and Scala 2.12 / kafka.admin.TopicCommandIntegrationTest.testDescribeUnderMinIsrPartitions(String).quorum=kraft
    Build / JDK 8 and Scala 2.12 / org.apache.kafka.streams.integration.LagFetchIntegrationTest.shouldFetchLagsDuringRestoration()
    Build / JDK 17 and Scala 2.13 / kafka.api.TransactionsTest.testBumpTransactionalEpoch(String).quorum=kraft
    Build / JDK 17 and Scala 2.13 / org.apache.kafka.controller.QuorumControllerTest.testBalancePartitionLeaders()
    Build / JDK 20 and Scala 2.13 / org.apache.kafka.connect.mirror.integration.IdentityReplicationIntegrationTest.testReplicateSourceDefault()
    Build / JDK 20 and Scala 2.13 / org.apache.kafka.connect.mirror.integration.MirrorConnectorsIntegrationExactlyOnceTest.testOffsetTranslationBehindReplicationFlow()
    Build / JDK 20 and Scala 2.13 / org.apache.kafka.streams.integration.NamedTopologyIntegrationTest.shouldRemoveOneNamedTopologyWhileAnotherContinuesProcessing()
```

File: core/src/main/java/kafka/log/remote/RemoteLogManager.java [hunk 2]
```diff
@@ -237,8 +239,8 @@ private void configureRLMM() {
         rlmmProps.put(KafkaConfig.LogDirProp(), logDir);
         rlmmProps.put("cluster.id", clusterId);
         endpoint.ifPresent(e -> {
-            rlmmProps.put("bootstrap.servers", e.host() + ":" + e.port());
-            rlmmProps.put("security.protocol", e.securityProtocol().name);
+            rlmmProps.put(REMOTE_LOG_METADATA_COMMON_CLIENT_PREFIX + "bootstrap.servers", e.host() + ":" + e.port());
+            rlmmProps.put(REMOTE_LOG_METADATA_COMMON_CLIENT_PREFIX + "security.protocol", e.securityProtocol().name);
```
Comment (chia7712) [inline core/src/main/java/kafka/log/remote/RemoteLogManager.java:243]: What happens if "this" cluster uses security and the "target" uses `PLAINTEXT`? Users might configure `remote.log.metadata.common.client.bootstrap.servers` but omit `remote.log.metadata.common.client.security.protocol`. It results in those client components will mistakenly try to connect to the `PLAINTEXT` cluster using `SSL/SASL`

**Pull request merged**